plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v4.metrics.test.ts

```diff
@@ -0,0 +1,169 @@
+  exportTimeoutMillis: 100,
+});
+
+otelTestingMeterProvider.addMetricReader(metricReader);
+
+import { registerInstrumentationTesting } from '@opentelemetry/contrib-test-utils';
+const instrumentation = registerInstrumentationTesting(
+  new MongoDBInstrumentation()
+);
+
+instrumentation.setMeterProvider(otelTestingMeterProvider);
+
+import { accessCollection, DEFAULT_MONGO_HOST } from './utils';
+
+import * as assert from 'assert';
+
+async function waitForNumberOfExports(
+  exporter: InMemoryMetricExporter,
+  numberOfExports: number
+): Promise<ResourceMetrics[]> {
+  if (numberOfExports <= 0) {
+    throw new Error('numberOfExports must be greater than or equal to 0');
+  }
+
+  let totalExports = 0;
+  while (totalExports < numberOfExports) {
+    await new Promise(resolve => setTimeout(resolve, 20));
+    const exportedMetrics = exporter.getMetrics();
+    totalExports = exportedMetrics.length;
+  }
+
+  return exporter.getMetrics();
+}
+
+describe('MongoDBInstrumentation-Metrics', () => {
+  // For these tests, mongo must be running. Add RUN_MONGODB_TESTS to run
+  // these tests.
+  const RUN_MONGODB_TESTS = process.env.RUN_MONGODB_TESTS as string;
+  let shouldTest = true;
+  if (!RUN_MONGODB_TESTS) {
+    console.log('Skipping test-mongodb. Run MongoDB to test');
+    shouldTest = false;
+  }
+
+  const HOST = process.env.MONGODB_HOST || DEFAULT_MONGO_HOST;
+  const PORT = process.env.MONGODB_PORT || '27017';
+  const DB_NAME = process.env.MONGODB_DB || 'opentelemetry-tests-metrics';
+  const COLLECTION_NAME = 'test-metrics';
+  const URL = `mongodb://${HOST}:${PORT}/${DB_NAME}`;
+
+  let client: mongodb.MongoClient;
+  let collection: mongodb.Collection;
+  beforeEach(function mongoBeforeEach(done) {
+    // Skipping all tests in beforeEach() is a workaround. Mocha does not work
+    // properly when skipping tests in before() on nested describe() calls.
+    // https://github.com/mochajs/mocha/issues/2819
+    if (!shouldTest) {
+      this.skip();
+    }
+    inMemoryMetricsExporter.reset();
+    done();
+  });
+
+  it('Should add connection usage metrics', async () => {
+    const result = await accessCollection(URL, DB_NAME, COLLECTION_NAME);
+    client = result.client;
+    collection = result.collection;
+    const insertData = [{ a: 1 }, { a: 2 }, { a: 3 }];
+    await collection.insertMany(insertData);
+    await collection.deleteMany({});
+    let exportedMetrics = await waitForNumberOfExports(
+      inMemoryMetricsExporter,
+      1
+    );
+
+    assert.strictEqual(exportedMetrics.length, 1);
+    let metrics = exportedMetrics[0].scopeMetrics[0].metrics;
+    assert.strictEqual(metrics.length, 1);
+    assert.strictEqual(metrics[0].dataPointType, DataPointType.SUM);
+
+    assert.strictEqual(
+      metrics[0].descriptor.description,
+      'The number of connections that are currently in state described by the state attribute.'
+    );
+    assert.strictEqual(metrics[0].descriptor.unit, '{connections}');
```

Comment:
looks like tav is failing here now :thinking: 
```suggestion
    assert.strictEqual(metrics[0].descriptor.unit, '{connection}');
```

Reply:
at least we have TAV. it helps us don't miss anything :)